core/src/test/java/com/linecorp/armeria/common/RequestContextWrapperTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.errorprone.annotations.MustBeClosed;
+
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+class RequestContextWrapperTest {
+
+    private static final class WrappedRequestContext extends RequestContextWrapper<RequestContext> {
+        private WrappedRequestContext(RequestContext delegate) {
+            super(delegate);
+        }
+
+        // Most wrappers will not want to push the delegate so we don't provide a default implementation of it.
+        @Override
+        @MustBeClosed
+        public SafeCloseable push() {
+            return delegate().push();
+        }
+    }
+
+    @Test
+    void wrapMatchesNormal() {
+        final RequestContext ctx = ServiceRequestContext.builder(HttpRequest.of(HttpMethod.GET, "/")).build();
+        // Use reflective comparison to handle added fields automatically.
+        assertThat(new WrappedRequestContext(ctx)).usingRecursiveComparison().ignoringFields("delegate")
+                                                  .isEqualTo(ctx);
```

Comment:
Is the behavior of `usingRecursiveComparison()` documented somewhere? How does it select the methods to call and compare the return values?

Reply:
Thanks! Found documentation at https://assertj.github.io/doc/#assertj-core-recursive-comparison and realized that I needed to set a configuration to allow it to extract Armeria properties